### DISCUSSION TITLE
Turn offloading on automatically if appropriate?

### DISCUSSION THREAD

**Discussion by CalculonPrime** [vladmandic/sdnext#3571] [Ideas]
The suggestion you made in the SD 3.5 thread worked, but it would be easier if this option were just set automatically on.

Is there a way to detect the graphics card, its memory, etc. and automatically set it so fewer adjustments are required to settings?

I'm pretty sure that ComfyUI is doing this automatically as I had no issue with SD 3.5 large and only have 16GB of VRAM.

**Comment by Disty0** [comment]
Model offload is turned on automatically with 8GB GPUs.
Sequential offload is turned on automatically with 4GB GPUs.

**Comment by vladmandic** [comment]
And while Balanced is probably best choice for larger models, its not always compatible so I'm reluctant to enable it by default.
Especially since compatibility is not specific to HW, but also to model and even model flavor such as quantization type it uses.

**Comment by CalculonPrime** [comment]
I have a 4060 Ti with 16GB VRAM, but somehow it was taking 45 seconds per iteration (so 15 minutes per rendering) with SD 3.5 Medium.

Once i set it to "balanced" the rendering was faster than ComfyUI, which takes 30 seconds to complete a rendering.

**Reply by vladmandic** [reply]
yes, i understand. i'm saying that sdnext supports a very broad range of platforms and models - and getting every single possibility perfectly right out of the box is impossible. this is not a software for users that are not willing to spend few minutes tuning.

**Reply by CalculonPrime** [reply]
It's your software, but note that automatic1111 became popular because it was so easy to use, requiring minimal tuning.

That project is no longer frequently updated now so those same people are looking for a replacement.  That could be your software.

**Reply by vladmandic** [reply]
that used to be the case in sd15 times. things are far more complex nowadays.
